widgets/src/main/java/org/pentaho/mantle/client/dialogs/folderchooser/FolderTree.java

```diff
@@ -648,8 +648,9 @@ private FolderTreeItem buildFolderTreeItem( @NonNull GenericFileTree fileTreeMod
 
     assert !fileModel.isGroupFolder() : "Folder tree item should not be mapped to group folder";
 
-    String name = fileModel.getName();
-    String title = fileModel.getTitleOrName();
+    //TODO BACKLOG-41091: Decoding would ideally be coming from the back-end (see also, BACKLOG-41089, BACKLOG-41229)
+    String name = GenericFileNameUtils.urlDecodePvfsFilePath( fileModel.getName() );
```

Comment:
I propose defining the method `GenericFile#getNameDecoded()` instead, and that `getTitle()` is already provided non-encoded (a clear bug of current implementation).
See related comment in: https://github.com/pentaho/pentaho-platform/pull/5647#pullrequestreview-2151052689.